### PR TITLE
[netcore] Throw ICE for Array.Copy(object[], enum[])

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/mono/mono/issues/14857
```csharp
Array src = new object[] { (object)1 };
Array dst = new DateTimeKind[] { DateTimeKind.Local };
src.CopyTo(dst, 0); // should throw InvalidCastException
```

### CODE REVIEW

**Comment by steveisok** [comment]
@monojenkins build failed

File: netcore/CoreFX.issues.rsp [hunk 1]
```diff
@@ -68,9 +60,6 @@
 ##  System.Diagnostics.TraceSource.Tests
 ####################################################################
 
-# requires precise gc (should be ignored in corefx)
```
Comment (vargaz) [inline netcore/CoreFX.issues.rsp]: These changes appear to be unrelated.

Comment (EgorBo) [reply netcore/CoreFX.issues.rsp]: @vargaz I grouped them (and added a few more) in order to remove them all once we get https://github.com/dotnet/corefx/pull/39176 (turns out I should mark more tests with that attribute)

**Comment by steveisok** [comment]
@monojenkins build failed
